src/emmy/calculus/derivative.cljc

```diff
@@ -441,17 +442,17 @@
 ;; instances.
 
 (def D
-  "Derivative operator. Takes some function `f` and returns a function
-  whose value at some point can multiply an increment in the arguments, to
-  produce the best linear estimate of the increment in the function value.
+  "Derivative operator. Takes some function `f` and returns a function whose value
+  at some point can multiply an increment in the arguments to produce the best
+  linear estimate of the increment in the function value.
 
-  For univariate functions, [[D]] computes a derivative. For vector-valued
+  For scalar-valued functions, [[D]] computes a derivative. For vector-valued
   functions, [[D]] computes
   the [Jacobian](https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant)
   of `f`.
 
-  The related [[Grad]] returns a function that produces a structure of the
-  opposite orientation as [[D]]. Both of these functions use forward-mode
+  The related [[emmy.env/Grad]] returns a function that produces a structure of
+  the opposite orientation as [[D]]. Both of these functions use forward-mode
```

Comment:
I think of Grad and D as the same thing, except Grad doesn't reverse tuple orientation because nobody learns about that in their first class in calculus of several variables, but that's years of exposure to SICM. I think we should keep that behavior, though, since in the ML community people expect to dot the gradient with the increment to form the directional derivative; so I think both choices {grad vs. D}, {forward vs. backward} could be made independently, IIUC?

Reply:
Yeah, I agree, these should be independently tuneable